src/catalog/src/durable/impls/persist.rs

```diff
@@ -119,19 +119,43 @@ impl PersistHandle {
         bootstrap_args: &BootstrapArgs,
         deploy_generation: Option<u64>,
     ) -> Result<Box<dyn DurableCatalogState>, CatalogError> {
-        let (is_initialized, upper) = self.is_initialized_inner().await;
+        let mut initialized_status = self.is_initialized_inner().await;
 
-        if !matches!(mode, Mode::Writable) && !is_initialized {
+        if !matches!(mode, Mode::Writable) && !initialized_status.is_initialized() {
             return Err(CatalogError::Durable(DurableCatalogError::NotWritable(
                 format!("catalog tables do not exist; will not create in {mode:?} mode"),
             )));
         }
 
+        // Listen cannot read any writes that happen at `Timestamp::minimum()`. So we initialize
+        // the shard with an empty write at `Timestamp::minimum()`.
+        if matches!(initialized_status, InitializedStatus::ShardUninitialized) {
+            let upper = initialized_status.upper();
+            let next_upper = upper.step_forward();
+            self.compare_and_append(Vec::new(), upper, next_upper)
+                .await?;
+            initialized_status = InitializedStatus::CatalogUninitialized;
+        }
```

Comment:
This approach ended up being much simpler than doing something like:
1. Initialize shard with initial catalog state at `Timestamp::minimum()`.
2. Get snapshot and start listening.

Initializing the shard with state would either need to take a different write path than all other writes (which is to use a transaction) or I would have had to do some serious refactoring.